include/xgboost/c_api.h

```diff
@@ -126,12 +126,29 @@ XGB_DLL int XGBGetGlobalConfig(char const **out_config);
 
 /*!
  * \brief load a data matrix
+ * \deprecated since 1.7.3
  * \param fname the name of the file
  * \param silent whether print messages during loading
  * \param out a loaded data matrix
  * \return 0 when success, -1 when failure happens
  */
 XGB_DLL int XGDMatrixCreateFromFile(const char *fname, int silent, DMatrixHandle *out);
+
+/*!
+ * \brief load a data matrix
+ * \param config JSON encoded parameters for DMatrix construction.  Accepted fields are:
+ *   - filename: The name of the file.
+ *   - silent (optional): Whether to print message during loading. Default to true.
+ *   - need_split (optional): Whether to split the file. Default to true in distributed mode, false
+ *     otherwise.
+ *   - data_split_mode (optional): Whether to split by row or column. If need_split is true, the
+ *     file is split accordingly; if false, this is only an indicator on how the file was split
+ *     beforehand. Default to row.
+ * \param out a loaded data matrix
+ * \return 0 when success, -1 when failure happens
+ */
+XGB_DLL int XGDMatrixCreateFromFileV2(char const *config, DMatrixHandle *out);
```

Comment:
Have you considered making it a proper API for URI? For instance `FromURI`? Since users actually need to use some of the URI formats like `file.txt?format=csv`. XGBoost/dmlc-core doesn't guess the format and is a source of bugs when users pass in only the file name.

Also, do you plan to introduce the `need_split` with other input sources as well? If not, we can make it an URI parameter and limit its use to this function. Otherwise, we need to have an additional parameter in all language bindings.

Reply:
Removed it as a parameter.